### ISSUE TITLE
Phone validation for Storefront.CustomerUpdateInput

### ISSUE THREAD

**Issue by tolkachovs** [Shopify/mobile-buy-sdk-ios#728]
In our application users can update their Shopify profile, that includes phone number. Now there are couple of problems with this:
1) if I send `nil`, for example if the user wants to remove his phone number completely, the mutation looks like this:
```
mutation {
  customerUpdate(customerAccessToken: "TOKEN", customer: {
    firstName: "Szymon", 
    lastName: "Tolkachov", 
    email: "szymon.tolkachov@proexe.pl"
  }) {
    customer {
      id
      firstName
      lastName
      email
      phone
      defaultAddress {
        id
        firstName
        lastName
        address1
        address2
        city
        zip
      }
    }
    userErrors {
      message
    }
  }
}
```
it doesn't send phone as a parameter;

2) if I send empty string, I receive an error stating "Invalid phone number".
3) in our application user doesn't have to provide country code, but Shopify reject phone numbers with out country code.

So to underline all of above, there is no way for a user to remove his phone number and there is no way to save phone with out providing a country code.

**Comment by dbart01** [comment]
1. This is valid point, currently the SDK doesn't support the use case. We're investigating the best approach to solve this.
2. What is the desired behaviour here? Shopify always returns a phone number with a country code. If a country code isn't provided on insertion / update, Shopify will need to make an assumption. Is it possible to just make that assumption on the client?

**Comment by tolkachovs** [comment]
1. Oh, I see.
2. I've tried it right now and if I send `phone:"881 632 215"` I got "_Phone is invalid_" response. I've also tried with out spaces and got the same response.
Overall, even if it would work like intended, it seems a bit confusing - you send one string and receive another. But I guess the decision is up to you. :)

**Comment by dbart01** [comment]
> it seems a bit confusing - you send one string and receive another

This is actually **not** the case. Since Shopify must always return a phone number with a country code, it also always expects that you provide a phone number with a country code.

```
send "+14161234567" -> Valid   -> Shopify returns "+14161234567
send "4161234567"   -> Invalid -> Shopify can't infer the country code
```
If you don't collect the country code from your users, you can make an educated guess on their likely phone number country code. However, I'd suggest simply letting the user enter their country code.

**Comment by dbart01** [comment]
Support for nullifying fields is now available in the [3.1 release](https://github.com/Shopify/mobile-buy-sdk-ios/tree/3.1).
